operators/benches/cache_concurrent.rs

```diff
@@ -0,0 +1,219 @@
+use std::sync::Arc;
+
+use futures::future::join_all;
+use geoengine_datatypes::primitives::{DateTime, SpatialPartition2D, SpatialResolution};
+use geoengine_datatypes::raster::RasterProperties;
+use geoengine_datatypes::{
+    primitives::{RasterQueryRectangle, TimeInterval},
+    raster::{Grid, RasterTile2D},
+    util::test::TestDefault,
+};
+use geoengine_operators::{engine::CanonicOperatorName, pro::cache::tile_cache::TileCache};
+use serde_json::json;
+
+enum Measurement {
+    Read(ReadMeasurement),
+    Write(WriteMeasurement),
+}
+
+struct ReadMeasurement {
+    read_query_ms: u128,
+}
+
+struct WriteMeasurement {
+    insert_query_ms: u128,
+    insert_tile_ms: u128,
+    finish_query_ms: u128,
+}
+
+async fn write_cache(tile_cache: &TileCache, op_name: CanonicOperatorName) -> WriteMeasurement {
+    let tile = RasterTile2D::<u8> {
+        time: TimeInterval::new_unchecked(1, 1),
+        tile_position: [-1, 0].into(),
+        global_geo_transform: TestDefault::test_default(),
+        grid_array: Grid::new([3, 2].into(), vec![1, 2, 3, 4, 5, 6])
+            .unwrap()
+            .into(),
+        properties: RasterProperties::default(),
+    };
+
+    let start = std::time::Instant::now();
+
+    let query_id = tile_cache
+        .insert_query::<u8>(op_name.clone(), &query_rect())
+        .await;
+
+    let insert_query_s = start.elapsed().as_millis();
+
+    let start = std::time::Instant::now();
+
+    tile_cache
+        .insert_tile(op_name.clone(), query_id, tile)
+        .await
+        .unwrap();
+
+    let insert_tile_s = start.elapsed().as_millis();
+
+    let start = std::time::Instant::now();
+
+    tile_cache
+        .finish_query(op_name.clone(), query_id)
+        .await
+        .unwrap();
+
+    let finish_query_s = start.elapsed().as_millis();
+
+    WriteMeasurement {
+        insert_query_ms: insert_query_s,
+        insert_tile_ms: insert_tile_s,
+        finish_query_ms: finish_query_s,
+    }
+}
+
+async fn read_cache(tile_cache: &TileCache) -> ReadMeasurement {
+    let op = op(0);
```

Comment:
Why is reading done only to one position?
I guess, also to see how future changes of the cache behave, we should do it similarly to the write stuff?

Reply:
We wanted to evaluate the impact of the locking so it shouldn't matter what entries we read, right?

Reply:
Yeah, but imagine we would change the implementation to have 8 locks for different segments. By just not reading only one element, we could cover those refactorings.

Reply:
I added a global counter and reading from random entries: 47b3ce7d934e3a0b24d08f53d2a10e3f156daf85